src/gei/Commands/GenerateScript/GenerateScriptCommandArgs.cs

```diff
@@ -45,6 +54,11 @@ public override void Validate(OctoLogger log)
                 {
                     throw new OctoshiftCliException("--ghes-api-url is invalid. Please check URL before trying again.");
                 }
+
+                if (UseGithubStorage)
+                {
+                    throw new OctoshiftCliException("--ghes-api-url must be specified when --use-github-storage is specified.");
+                }
```

Comment:
this will raise in a happy path scenario! what you need is to check `GhesApiUrl.IsNullOrWhitespace() && UseGithubStorage` after the aws bucket name validation.